### PR TITLE
Fixes #117 multipart response message

### LINKED ISSUE #117: Multipart response message fix
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the CDP4 Web Services
- [x] I have searched [open](https://github.com/RHEAGROUP/CDP4-WebServices-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/CDP4-WebServices-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
Message format for a multipart message is not according to [RFC 1314](https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html).
The end point is not there and the main message Content-Type is not according to standard.

Please, addd the end point and the correct content type.

### PR DESCRIPTION
**Pull request by lxatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-WebServices-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-SDK [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-WebServices-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
Fixes multipart resonse message according to [rfc1341](https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html)

### CODE REVIEW

File: CDP4WebServices.API/Modules/10-25/ApiBase.cs [hunk 1]
```diff
@@ -867,19 +881,31 @@ protected string GenerateRandomToken()
...
+                this.AddMultiPartMimeEndpoint(targetStream);
             }
             finally
             {
                 this.FileArchiveService.DeleteFileStructureWithArchive(folderPath);
             }
         }
 
+        /// <summary>
+        /// //add ending line according to https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html
```
Comment (alexatrhea) [inline CDP4WebServices.API/Modules/10-25/ApiBase.cs:900]: Some rewording and //

File: CDP4WebServices.API/Services/Headers/HeaderInfoProvider.cs [hunk 2]
```diff
@@ -111,12 +111,25 @@ public void RegisterResponseHeaders(Response response)
...
+        /// Register the multipart CDP4 content-type header to the passed in response.
+        /// </summary>
+        /// <param name="response">
+        /// The nancy response.
+        /// </param>
+        /// <param name="boundaryString">
+        /// The boundary text in a Multipart MIME message <see href="https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html"/>
+        /// </param>
+        public void RegisterMultipartResponseContentTypeHeader(Response response, string boundaryString) => 
+            response.Headers.Add(this.ContentTypeHeader, $"multipart/mixed; ecss-e-tm-10-25; version=1.0.0; boundary={boundaryString}");
```
Comment (alexatrhea) [inline CDP4WebServices.API/Services/Headers/HeaderInfoProvider.cs:131]: Lets keep on-liners to Property getters please.

File: CDP4WebServices.API/config_debug.json [hunk 3]
```diff
@@ -16,8 +16,8 @@
     "DatabaseManage": "cdp4manage",
     "StatementTimeout": 180,
     "LogSqlCommands": false,
-    "IsDbSeedEnabled": true,
-    "IsDbRestoreEnabled": true
+    "IsDbSeedEnabled": false,
```
Comment (alexatrhea) [inline CDP4WebServices.API/config_debug.json:19]: Was this intentional?

**Comment by mmilitaru** [comment]
Please change copyright notice to in accordance with AGPL license.

File: CDP4WebServices.API/config_debug.json (hunk 3 above)
Comment (lxatrhea) [reply CDP4WebServices.API/config_debug.json:19]: Yes, because this has overwritten my local test database once and I don't want that to happen again.

Comment (samatrhea) [reply CDP4WebServices.API/config_debug.json:19]: there is a "test" config that should be used which has these set to `true`

File: CDP4WebServices.API/Services/Headers/HeaderInfoProvider.cs [hunk 4]
```diff
@@ -111,12 +130,27 @@ public void RegisterResponseHeaders(Response response)
...
+        /// </summary>
+        /// <param name="response">
+        /// The nancy response.
+        /// </param>
+        /// <param name="boundaryString">
+        /// The boundary text in a Multipart MIME message <see href="https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html"/>
+        /// </param>
+        public void RegisterMultipartResponseContentTypeHeader(Response response, string boundaryString)
+        {
+            response.Headers.Add(this.ContentTypeHeader, $"multipart/mixed; ecss-e-tm-10-25; version=1.0.0; boundary={boundaryString}");
```
Comment (samatrhea) [inline CDP4WebServices.API/Services/Headers/HeaderInfoProvider.cs:151]: please have a look at the the `RegisterResponseHeaders`, i think this response headers also need the following to stay inline with eachother

```
response.Headers.Add(this.Cdp4ServerHeader, this.Cdp4ServerVersion);
response.Headers.Add(this.Cdp4CommonHeader, this.Cdp4CommonVersion);
```

Comment (lxatrhea) [reply CDP4WebServices.API/Services/Headers/HeaderInfoProvider.cs:151]: That is true. 
These two headers are added in a generic method call higher up the callstack.
So no problem here.

File: CDP4WebServices.API/config_debug.json (hunk 3 above)
Comment (lxatrhea) [reply CDP4WebServices.API/config_debug.json:19]: That's why I set them to false here, in the "debug" config.

**Comment by lxatrhea** [comment]
> 
> 
> looking at this class we may be able to refactor it a bit more to improve performance. Seems like now we do reflection each time, which i think we only need once. Perhaps we make a new task for this as a future improvement.

Fixing is faster than writing an issue. I'll do it now.

File: CDP4WebServices.API/Modules/10-25/ApiBase.cs [hunk 5]
```diff
@@ -867,19 +900,31 @@ protected string GenerateRandomToken()
...
             {
                 this.FileArchiveService.DeleteFileStructureWithArchive(folderPath);
             }
         }
 
+        /// <summary>
+        /// add ending line according to https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html
+        /// </summary>
+        /// <param name="targetStream">The <see cref="Stream"/>where to add the end poit to</param>
+        private void AddMultiPartMimeEndpoint(Stream targetStream)
```
Comment (samatrhea) [inline CDP4WebServices.API/Modules/10-25/ApiBase.cs:922]: can we make the parameter a `ref` parameter to make it obvious we are chaining the parameter `targetStream` in this mehtod?

**Pull request merged**